### PR TITLE
docs: update docs for multiple dtrain single node [DET-3822]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This adds docs for the single node multiple dtrain work.

## Test Plan
- [x] build and read documentation.

## Commentary (optional)
This gets a bit wordy in places; size of the subset of experiment/cluster configurations that require a job to set `resources.slots_per_trial` to be a multiple of agent size is smaller but only because there are more constraints, and enumerating those constraints takes more text. Once we don't hardcode ports, it should get simpler, again.

## Checklist

- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/how-to/distributed-training.txt [hunk 1]
```diff
@@ -87,11 +87,21 @@ S3. In this example, the S3 bucket name is configured via a field
...
-To use distributed training, ``slots_per_trial`` must be set to a multiple
-of the number of GPUs per machine. For example, if you have a cluster of machines with
-8 GPUs each, you should set ``slots_per_trial`` to a multiple of 8, such as 8, 16, or 24.
-This ensures that the full network and interconnect bandwidth are available to
-multi-GPU workloads and results in better performance.
+When using distributed training, there are different scheduler constraints on the value
+``slots_per_trial`` can take depending on the default network mode and whether or not it can fit on
+a single agent.
+
+- If ``slots_per_trial`` is smaller than the number of slots on any available agents and
```
[the remark this answers is not in the document]
Comment (stoksc) [reply docs/how-to/distributed-training.txt:94]: Yeah, I agree its confusing, splitting it out like above makes it a little easier to understand. I'll make that change.

> Also, when we are talking about the relationship between slots_per_trial and the # of slots on any agent -- suppose I have one agent with 16 GPUs and 10 agents with 8 GPUs each. If I submit an 8 GPU job, is it possible that it gets spread over multiple agents?

No, in that case is should only ever get scheduled onto a single agent. If an experiment is spread over multiple agents, it will always take up entire agents.

[the remark this answers is not in the document]
Comment (stoksc) [reply docs/how-to/distributed-training.txt]: @neilconway what do you think now?

[the remark this answers is not in the document]
Comment (neilconway) [reply docs/how-to/distributed-training.txt]: @stoksc I made some edits and pushed to the branch -- lmk what you think.

[the remark this answers is not in the document]
Comment (stoksc) [reply docs/how-to/distributed-training.txt]: It looks good to me now. I think I approach docs trying to rewrite (change) as little as possible and I think it's leading my astray; these look much better and they're mostly rewritten. I'll let @aaron276h take the last pass tomorrow, though, but I would approve this.

File: docs/how-to/distributed-training.txt [hunk 2]
```diff
@@ -83,30 +83,49 @@ S3. In this example, the S3 bucket name is configured via a field
...
-of the number of GPUs per machine. For example, if you have a cluster of machines with
-8 GPUs each, you should set ``slots_per_trial`` to a multiple of 8, such as 8, 16, or 24.
-This ensures that the full network and interconnect bandwidth are available to
-multi-GPU workloads and results in better performance.
+The Determined master takes care of scheduling distributed training jobs
+automatically, ensuring that all of the compute resources required for a job are
+available before the job itself is launched. Users should be aware of the
+following details about scheduler behavior when using distributed training:
+
+- When using Docker host-mode networking (e.g., when
```
Comment (aaron276h) [inline docs/how-to/distributed-training.txt:96]: non-blocking: I would re-order these points to be in the inverse order of what they are now, since I expect most customers would be running in docker networking and this top point isn't applicable to them.

File: docs/how-to/distributed-training.txt [hunk 3]
```diff
@@ -83,30 +83,49 @@ S3. In this example, the S3 bucket name is configured via a field
...
 -------------------
 
-To use distributed training, ``slots_per_trial`` must be set to a multiple
-of the number of GPUs per machine. For example, if you have a cluster of machines with
-8 GPUs each, you should set ``slots_per_trial`` to a multiple of 8, such as 8, 16, or 24.
-This ensures that the full network and interconnect bandwidth are available to
-multi-GPU workloads and results in better performance.
+The Determined master takes care of scheduling distributed training jobs
+automatically, ensuring that all of the compute resources required for a job are
+available before the job itself is launched. Users should be aware of the
```
Comment (aaron276h) [inline docs/how-to/distributed-training.txt:93]: not exactly true for the k8s RP but probably not worth making this even more difficult for users to understand.

**Comment by stoksc** [comment]
Going to hold on this, since @aaron276h thinks its possible to do DET-3821 more quickly that originally planned.

File: docs/how-to/distributed-training.txt [hunk 4]
```diff
@@ -83,30 +83,42 @@ S3. In this example, the S3 bucket name is configured via a field
...
+following details about scheduler behavior when using distributed training:
+
+- If ``slots_per_trial`` is smaller than the number of slots on a
+  single agent, Determined will consider scheduling multiple distributed training
+  jobs on a single agent. This is designed to improve utilization and to allow
+  multiple small training jobs to run on a single agent. For example, an agent
+  with 8 GPUs could be assigned two 4-GPU jobs, or four 2-GPU jobs.
+- Otherwise, if ``slots_per_trial`` is greater than the number of slots on a single agent,
```
Comment (neilconway) [inline docs/how-to/distributed-training.txt:101]: If we're going to phrase it this week, the previous "If ..." should be "smaller than or equal to", not "smaller than".

File: docs/faq.txt [hunk 5]
```diff
@@ -97,14 +97,19 @@ Distributed Training
...
-cluster.
-
-See :ref:`multi-gpu-training` for more details.
+If :ref:`slots_per_trial <exp-config-resources-slots-per-trial>` is greater than
+the number of slots on a single agent, Determined will schedule it over multiple
+machines. When scheduling a multi-machine distributed training job, Determined
+requires  that the job uses all of the slots (GPUs) on an agent. For example, in
+a cluster that consists of 8-GPU agents, an experiment with
+:ref:`slots_per_trial <exp-config-resources-slots-per-trial>` set to ``12`` will
+never be scheduled, and will instead wait indefinitely. The :ref:`distributed
```
Comment (neilconway) [inline docs/faq.txt:106]: Remove the comma after "scheduled"
